lib/ansible/collections/list.py

```diff
@@ -10,12 +10,27 @@
 
 from ansible.collections import is_collection_path
 from ansible.module_utils._text import to_bytes
+from ansible.module_utils.six import string_types
 from ansible.utils.collection_loader import AnsibleCollectionConfig
+from ansible.utils.collection_loader._collection_finder import AnsibleCollectionRef, _get_collection_name_from_path
 from ansible.utils.display import Display
 
 display = Display()
 
 
+def validate_collection_path(collection_path):
+    """Ensure a given path ends with 'ansible_collections'
+
+    :param collection_path: The path that should end in 'ansible_collections'
+    :return: collection_path ending in 'ansible_collections' if it does not already.
+    """
+
+    if os.path.split(collection_path)[1] != 'ansible_collections':
```

Comment:
Since this is used for more than ansible-galaxy now it should probably be made consistent with https://github.com/ansible/ansible/blob/51f6d129cbb30f42c445f7e2fecba68fe02d6f85/lib/ansible/utils/collection_loader/_collection_finder.py#L62 - either `ansible_collections` should always be joined here (my preference) or the collection loader should only conditionally join it.

Currently someone may install a collection with
```
# ansible.cfg
[defaults]
collections_paths = ./my/collections/ansible_collections
```
And when they use the same config to use content in the collection, ansible will look for it at `./my/collections/ansible_collections/ansible_collections` and not find it.

Reply:
added that instance to function, also moved so it is in finder